layerfilter.py

```diff
@@ -51,29 +51,42 @@ def __init__(
         self.current_layer_name = ''
 
         self.initialize_state()
-
-    def get_layer_names(self):
-        # reads through the root to parse out the layers present in the file
-        if '/OCProperties' not in self.pdf.Root.keys():
-            return None 
-
-        names = [str(oc.Name) for oc in self.pdf.Root.OCProperties.OCGs]
-        ordered_names = []
-        for o in self.pdf.Root.OCProperties.D.Order:
+    
+    @staticmethod
+    def _fix_utf16(string):
+        new_string = string.replace('\x00', '')
+        if new_string.startswith('ÿþ'):
+            new_string = new_string[2:]
```

Comment:
My concern with this workaround is we only have the one test case, but it seems to be a pretty rare issue anyway.

Reply:
Yeah, you're right... I think we should still report this to pikepdf to find the root cause at some point.